src/modules/ws/websocket.ts

```diff
@@ -0,0 +1,48 @@
+import SendMessageValidator from './message_validator';
+
+interface WebSocketConnection {
+  ws: WebSocket;
+  sendMessage(key: string, data: Record<string, unknown>): void;
+}
+
+const WEBSOCKET_URL = process.env.REACT_APP_WEBSOCKETURL || '';
+
+/**
+ * Interface that the client can use for Websocket Connection.
+ */
+class WebSocketConnection {
+  public constructor(url: string = WEBSOCKET_URL) {
+    this.ws = new WebSocket(url);
+    this.ws.onopen = (event) => {
+      console.log(event);
+      this.ws.send('');
+    };
+
+    this.ws.onclose = (event) => {
+      console.log(event);
+    };
+
+    this.ws.onerror = (error) => {
+      console.log(error);
+    };
+
+    this.ws.onmessage = (event) => {
+      console.log(event);
+    };
+  }
+}
+
+/**
+ * Sends a message to the backend if the message is validated.
+ */
+WebSocketConnection.prototype.sendMessage = function sendMessage(key: string, data: Record<string, unknown>): boolean {
```

Comment:
I've never seen a function bound to the prototype, could you elaborate on using prototype to hold function?

Reply:
prototype basically holds fields and methods of a js class, but the fields and methods can be modified even after a class has been instantiated. You can try it in the browser lol. Note: ts classes are eventually compiled to prototypes, but it is easier to read and safter (can add access modifiers, public private etc.)

Reply:
yeah but why would you want to modify a method after a class has been instantiated? and for fields isn't that what gettings and setters are for?

Reply:
Oh, it allows for the interface. I see what you did.

Reply:
That's interesting. What is the value of such an interface?

Reply:
Interface allows for control of the object that we have created. User would not have direct access to the websocket object and will need to use our interface to use its functionality. https://refactoring.guru/design-patterns/proxy